lnwire/lnwire.go

```diff
@@ -589,6 +633,19 @@ func ReadElement(r io.Reader, element interface{}) error {
 		if _, err := io.ReadFull(r, *e); err != nil {
 			return err
 		}
+
+	case *WarningData:
```

Comment:
It would be nicer but we cannot because the `switch` statement is casting to the data type. Even if the underlying struct looks the same we cannot access to its properties until we cast it again. Same case as `ShortChanIDEncoding` and `uint8`